Improve selection of file extensions for auto-mode-alist

rrthomas opened the pull request:
Remove .xqx, as that is for XQueryX, i.e. XQuery as XML.

Add .xquery, .xq, .xqm, .xql (as listed on the Wikipedia page for XQuery).

rrthomas commented:
Thanks for xquery-mode!

rrthomas commented:
Note that two Travis builds failed because cask is no longer supported on Emacs < 24.5; I don't understand why the 24.5 build failed (cask was trying to install emacs-25.1???); the 25.1 build passed. Perhaps you can update the versions of Emacs you test against anyway?

proofit404 commented on xquery-mode.el:
```diff
@@ -316,7 +314,7 @@
 
 ;;;###autoload
-(add-to-list 'auto-mode-alist '(".xq[erxy]\\'" . xquery-mode))
+(add-to-list 'auto-mode-alist '("\\.\\(?:xquery\\|xq[elmry]?\\)\\'" . xquery-mode))
```
I can't find `xqe` & `xqr` extensions on Wikipedia page https://en.wikipedia.org/wiki/XQuery

![Untitled](https://user-images.githubusercontent.com/1862725/120713249-509f7f00-c4ca-11eb-91d2-2d1e969e7212.png)

rrthomas commented:
@proofit404, that's correct, I added the extensions from Wikipedia that were not already supported, removed `.xqx`, and kept the other pre-existing extensions, which I checked by searching online were in use.

proofit404 commented:
@m-g-r If this change does not break any of existed workflows, I don't mind to merge it.

Thanks!
